libtcmu_config.c

```diff
@@ -356,11 +375,64 @@ static void tcmu_parse_options(struct tcmu_config *cfg, char *buf, int len)
 
 }
 
-int tcmu_load_config(struct tcmu_config *cfg, const char *path)
+int tcmu_reload_config(struct tcmu_config *cfg)
 {
 	char buf[TCMU_MAX_CFG_FILE_SIZE];
 	int fd, len;
 
+	fd = open(cfg->path, O_RDONLY);
+	if (fd < 0) {
+		tcmu_err("Failed to open file '%s'\n", cfg->path);
+		return -1;
+	}
+
+	len = tcmu_read_config(fd, buf, TCMU_MAX_CFG_FILE_SIZE);
+	close(fd);
+	if (len < 0) {
+		tcmu_err("Failed to read file '%s'\n", cfg->path);
+		return -1;
+	}
+
+	buf[len] = '\0';
+
+	tcmu_parse_options(cfg, buf, len);
+
+	return 0;
+}
+
+static void *dyn_config_thread_start(void *arg)
+{
+	struct tcmu_config *cfg = arg;
+	struct stat m_stat;
+	int ret;
+
+	while (1) {
+		sleep(10);
+
+		ret = stat(cfg->path, &m_stat);
+		if (ret == -1) {
```

Comment:
Yes, for linux the inotify is a better way to do this. 

Not sure will we adapt this to other unix os, such as Mac OS, in future?